### ISSUE TITLE
[FR] Would you consider adding a programmatic API?

### ISSUE THREAD

**Issue by abravalheri** [tox-dev/pyproject-fmt#3]
Thank you again for the nice and useful package @gaborbernat!

Would you consider a PR that adds an API for programmatic use of `pyproject_fmt` (as a library instead of a CLI command)?

**Is your feature request related to a problem? Please describe.**

I currently maintain the [`ini2toml`](https://ini2toml.readthedocs.io/en/latest/) package whose main objective is to assist people migrating from `setup.cfg` to something `pyproject.toml` (not only for setuptools, but also for other tools like `mypy`, `isort`, etc...).

I would like to use `pyproject-fmt` in the generated TOML string.

**Describe the solution you'd like**

It would be nice to be able to use `pyproject-fmt` in the following way:

```python
from pyproject_fmt import format

text = """
[project]
name = "myproj"
...
"""

formatted_text = format(text, indent=2)
```
(any alternate name suggestion is appreciated, I am bad at naming)

**Describe alternatives you've considered**

Currently I am using something similar to the following:

```python
from tempfile import NamedTemporaryFile

from pyproject_fmt.cli import PyProjectFmtNamespace
from pyproject_fmt.formatter import format_pyproject

with NamedTemporaryFile(mode="w", encoding="utf-8", delete=False) as tmp:
   tmp.write(text)
   tmp.close()
   opts = PyProjectFmtNamespace(
       pyproject_toml=Path(tmp.name), stdout=True, indent=2
   )
   formatted_text = format_pyproject(opts)
```

But I seriously suspect those are all private APIs (also having to use a temporary file to modify a string in memory is not very handy).

**Comment by gaborbernat** [comment]
I'd be ok with this. We can re-export both format_pyproject and PyProjectFmtNamespace and root level. If we do this though I'd like to add sphinx documentation generation for the public part.

**Comment by gaborbernat** [comment]
Resolved by #9
